src/test/integration/integrationtests.py

```diff
@@ -144,6 +147,125 @@ def test_jobs_by_osu_id(self):
         self.assertEqual(
             utils.get_person_by_osu_id(not_valid_osu_id).status_code, 404)
 
+    def test_post_job_not_found(self):
+        # expect 404 if osuID is not valid
+        self.assertEqual(
+            utils.post_job_by_osu_id(not_valid_osu_id, {}).status_code, 404)
+
+    def test_bad_post_job(self):
+        # "data" field is empty
+        self.validate_bad_response({"data": {}}, "No job object provided")
+
+        # dates are not in ISO8601 format
+        self.validate_bad_job_post(test_job, "effectiveDate", "badDate",
+                                   "Make sure dates are in ISO8601 format")
+
+        # required field is missing
+        required_fields = ["positionNumber", "beginDate", "supervisorOsuID",
+                           "supervisorPositionNumber", "effectiveDate"]
+        for field in required_fields:
+            self.validate_bad_job_post(test_job, field, None,
+                                       field + " is required")
+
+        # positive field has negative value
+        positive_fields = ["hourlyRate", "hoursPerPay", "assignmentSalary",
+                           "annualSalary", "paysPerYear"]
+        for field in positive_fields:
+            self.validate_bad_job_post(test_job, field, "-1",
+                                       field + " cannot be a negative number")
+
+        # invalid fields
+        invalid_fields = [
+            {"supervisorOsuID": "Supervisor OSU ID does not exist"},
+            {"positionNumber": "not a valid position number for the given "
+                               "begin date"},
+            {"locationID": "not a valid location ID"},
+            {"timesheetOrganizationCode": "not a valid organization code"}
+        ]
+        for field, message in invalid_fields:
```

Comment:
This line returned this error: 
```
======================================================================
ERROR: test_bad_post_job (__main__.TestStringMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "integrationtests.py", line 185, in test_bad_post_job
    for field, message in invalid_fields:
ValueError: not enough values to unpack (expected 2, got 1)

----------------------------------------------------------------------
```

Maybe `invalid_fields` should just be one dict instead of an array of dicts?

Reply:
Ah right, thanks for the catch